File: app/src/main/java/com/wang/example/msg/bean/OrderBean.java
package com.wang.example.msg.bean;

import androidx.annotation.NonNull;

import com.alibaba.fastjson.JSON;
import com.wang.container.adapter.IContainerItemAdapter;
import com.wang.example.msg.adapter.PaySuccessOrderAdapter;
import com.wang.example.msg.adapter.WaitPayOrderAdapter;

public class OrderBean extends BaseMsgBean {
    public OrderDataEntity orderInfo;

    protected OrderBean(BaseMsgBean oldBean) {
        super(oldBean);
        orderInfo = JSON.parseObject(oldBean.data, OrderDataEntity.class);
    }

    public static class OrderDataEntity {
        public int orderType;//1未支付，2已支付
        public String orderNo, orderName;
        public OtherOrderData otherOrderData;

        public static class OtherOrderData {
            public String emsNo;
        }
    }

    @NonNull
    @Override
    public Class<? extends IContainerItemAdapter> getBindAdapterClass() {
        switch (orderInfo.orderType) {
            case 1:
                return WaitPayOrderAdapter.class;
            case 2:
                return PaySuccessOrderAdapter.class;
        }
        return super.getBindAdapterClass();
    }
}

File: app/src/main/java/com/wang/example/msg/bean/OrderBean.kt
package com.wang.example.msg.bean

import com.google.gson.Gson
import com.wang.container.adapter.BaseContainerItemAdapter
import com.wang.example.msg.adapter.PaySuccessOrderAdapter
import com.wang.example.msg.adapter.WaitPayOrderAdapter

class OrderBean(oldBean: BaseMsgBean) : BaseMsgBean(oldBean) {
    var orderInfo: OrderDataEntity

    data class OrderDataEntity(
        val orderType: Int = 0, //1未支付，2已支付
        val orderNo: String = "",
        val orderName: String = "",
        val otherOrderData: OtherOrderData = OtherOrderData()
    ) {
        data class OtherOrderData(val emsNo: String = "")
    }

    override fun getBindAdapterClass(): Class<out BaseContainerItemAdapter<*>> {
        when (orderInfo.orderType) {
            1 -> return WaitPayOrderAdapter::class.java
            2 -> return PaySuccessOrderAdapter::class.java
        }
        return super.getBindAdapterClass()
    }

    init {
        orderInfo = Gson().fromJson(oldBean.data, OrderDataEntity::class.java)
    }
}